lib/components/map/bounds-updating-overlay.js

```diff
@@ -69,7 +69,31 @@ class BoundsUpdatingOverlay extends MapLayer {
     const oldIntermediate = oldProps.query && oldProps.query.intermediatePlaces
     const newIntermediate = newProps.query && newProps.query.intermediatePlaces
     const intermediateChanged = !isEqual(oldIntermediate, newIntermediate)
-    if (
+
+    // Also refit map if itineraryView prop has changed.
+    const itineraryViewChanged = oldProps.itineraryView !== newProps.itineraryView
+
+    if (itineraryViewChanged) {
+      // If itineraryView has changed,
+      // force a resize of the map before re-fitting the active itinerary or active leg,
+      // and do that after a delay to that canvas heights have stabilized in the DOM.
+      setTimeout(() => {
+        map.invalidateSize(true)
+
+        if (newProps.itinerary) {
+          if (newProps.activeLeg !== null) {
+            // Fit to active leg if set.
+            map.fitBounds(
+              getLeafletLegBounds(newProps.itinerary.legs[newProps.activeLeg]),
+              { padding }
+            )
+          } else {
+            // Fit to whole itinerary otherwise.
+            map.fitBounds(newItinBounds, { padding })
+          }
+        }
+      }, 250)
```

Comment:
Non-blocking: This whole block might be nice in a `fitMapToItinerary` method.

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/339/commits/b738bd20222e0614ab23025d94634b769981bac7.